### PR TITLE
Add upgrade-notes.md file

### PR DESCRIPTION
**Pull request by elemoine**
This file includes upgrade notes for changes to the library that require changes to application code. This file is used by the release manager to fill the "Upgrade notes" section in the v3.x.y.md changelog file.

Ping @ahocevar.

### CODE REVIEW

File: changelog/upgrade-notes.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## Upgrades notes
```
Comment (bartvde) [inline changelog/upgrade-notes.md]: Upgrade notes?

Comment (elemoine) [inline changelog/upgrade-notes.md]: Changed. Thanks @bartvde.

**Comment by ahocevar** [comment]
Not sure I understand what was decided, but shouldn't there be a heading that states which upgrade path the notes are relevant for? Something like
```
## v3.3.x
```

**Comment by elemoine** [comment]
My plan was to accumulate upgrade notes in this file and have the release manager extract the notes that are relevant to his release.

**Comment by tschaub** [comment]
Sounds good to me.

**Comment by ahocevar** [comment]
This will work for the release manager and anyone who can do a `git blame`, but will look cryptic for anyone else. But if others think that the way it is now is the way to go, I don't mind.

**Comment by gberaudo** [comment]
The release managers may prepend the release tag (and date?) each time a
release occur. This way changes will be separated by release.

**Comment by elemoine** [comment]
> The release managers may prepend the release tag (and date?) each time a release occur. This way changes will be separated by release.

Yep!

**Comment by ahocevar** [comment]
Sounds good to me. @elemoine, please merge and update the release procedure doc to add this step.

**Comment by elemoine** [comment]
> update the release procedure doc to add this step.

Done.

**Pull request merged**